File: pe001/pe001-naive.go
/*
Project Euler, Problem 1:

Find the sum of all the multiples of 3 or 5 below 1000.

https://projecteuler.net/problem=1
*/

package main

import "fmt"

var divisors = []int{3, 5}
var max_number = 999

func main() {
    result := 0
    for i := 2; i <= max_number; i++ {
        for _, d := range divisors {
            if (i % d == 0) {
                result += i
                break
            }
        }
    }
    fmt.Println(result)
}

File: pe001/pe001-naive.py
#!/usr/bin/python

"""
Project Euler, Problem 1:

Find the sum of all the multiples of 3 or 5 below 1000.

https://projecteuler.net/problem=1
"""

divisors = [3, 5]
max_number = 999

def main():
    result = 0
    for i in range(2, max_number + 1):
        if min([ i % d for d in divisors ]) == 0:
            result += i   
    return str(result)

if __name__ == "__main__":
    print(main())
